src/folder/FolderController.ts

```diff
@@ -0,0 +1,193 @@
+            const schema = yup.object<FolderCreate>().shape({
+                title: yup.string().required(),
+                boardId: yup.string().required(),
+                description: yup.string().required(),
+            });
+            return schema.validate(body);
+        });
+
+        if (validationError) {
+            this.setStatus(400);
+            return validationError;
+        }
+
+        const user = await User.findOneBy({ id: req.ctx.state.user.id });
+
+        if (!user) {
+            this.setStatus(404);
+            return `No user found with id ${req.ctx.state.user.id}`;
+        }
+
+        return Folder.create({
+            title: folder.title,
+            description: folder.description,
+            boardId: folder.boardId,
+            tasks: [],
+        }).save();
+    }
+
+    @SuccessResponse(200, "OK")
+    @Response<yup.ValidationError>(400)
+    @Put(`/{id}`)
+    public async update(@Path() id: string, @Body() body: FolderUpdate) {
+        const [validationError, folder] = await tri(() => {
+            const schema = yup.object<FolderUpdate>().shape({
+                title: yup.string(),
+                boardId: yup.string(),
+                description: yup.string(),
+            });
+            return schema.validate(body);
+        });
+
+        if (validationError) {
+            this.setStatus(400);
+            return validationError;
+        }
+
+        return Folder.save({ id, ...folder });
+    }
+
+    @SuccessResponse(200, "OK")
+    @Response<string>(404, "No board found with id <id>")
+    @Delete(`/{id}`)
+    public async delete(@Path() id: string) {
+        const folder = await Folder.findOneBy({ id });
+
+        if (!folder) {
+            this.setStatus(404);
+            return `No board found with id ${id}`;
+        }
+
+        const { boardId, boardIndex } = folder;
+        const count = await Folder.countBy({ boardId });
+
+        await folder.remove();
+
+        if (boardIndex === 0) {
+            await Folder
+                .getRepository()
+                .decrement(
+                    { boardId },
+                    "boardIndex",
+                    1
+                );
+        }
+        else if (boardIndex > 0 && boardIndex < count) {
+            await Folder
+                .getRepository()
+                .decrement(
+                    { boardId, boardIndex: MoreThan(boardIndex) },
+                    "boardIndex",
+                    1
+                );
+        }
+    }
+
+    @SuccessResponse(200, "OK")
+    @Response<string>(404, "No folders found for board <id>")
+    @Get(`/board/{boardId}`)
+    public async boardFolders(@Path() boardId: Board["id"]) {
+        const folders = Folder.find({
+            where: { boardId },
+            order: { boardIndex: "asc" }
+        });
+
+        if (!folders) {
+            this.setStatus(404);
+            return `No folders found for board ${boardId}`;
+        }
```

Comment:
I don't think this is needed since if there are no folders in a board an empty array will be returned. So the response will never be `null`